src/execution/execution.cc

```diff
@@ -343,6 +343,41 @@ V8_WARN_UNUSED_RESULT MaybeHandle<Object> Invoke(Isolate* isolate,
       DCHECK(!params.IsScript());
     }
 #endif
+
+    if (recordreplay::AreEventsDisallowed() &&
+        !recordreplay::HasDivergedFromRecording() &&
+        function->shared().IsUserJavaScript() &&
+        function->shared().HasSourceCode()) {
+      // [RUN-1621] User JS should not get executed non-deterministically,
+      // unless we have paused.
+
+      // Get script name.
+      int script_id = -1;
+      std::string scriptName = "";
+      if (function->shared().script().IsScript()) {
+        Script script = Script::cast(function->shared().script());
+        Object name_or_url = script.GetNameOrSourceURL();
+        script_id = script.id();
+        if (name_or_url.IsString()) {
+          std::unique_ptr<char[]> name_raw =
+              String::cast(name_or_url).ToCString();
+          scriptName = name_raw.get();
+        }
+      }
+
+      // Always allow Replay-internal scripts.
+      if (scriptName != "record-replay-internal") {
+        // Print log and prevent execution.
+        recordreplay::Print(
```

Comment:
If we're going to leave this Print in it should be converted to a warning --- in general we don't want this to be hit, callers should be doing the appropriate checks to avoid running user JS when events are disallowed.  Warnings will be surfaced in honeycomb etc. by our observability but raw print statements will not be.

Reply:
Ok: I'll change it back to `Warning` and also add them to `IgnoredWarnings`, so we will see them and they won't override other, more important warnings.

I don't want this to spam us because the main culprit here is CDP and `ValueMirror` calling `Object::GetProperty` without checks and balances in a few places. And trying to figure out whether or not `Object::GetProperty` will call into user JS is a bit tricky (could be accessors, interceptors or proxy behavior). But maybe that'll be the required solution. Added it to `RUN-1616`.